evap/student/views.py

```diff
@@ -24,6 +28,79 @@
 SUCCESS_MAGIC_STRING = "vote submitted successfully"
 
 
+@dataclass
+class GlobalRewardWithProgress:
```

Comment:
Maybe this class would be better placed under the `GlobalRewards` namespace, for example `GlobalRewards.Reward` or `GlobalRewards.Step`?

Reply:
I'm generally open for restructuring (and the static factory method makes sense to me), but I don't think `.Reward` or `.Step` are better names than `WithProgress`.

Reply:
I called it `GlobalRewards.RewardProgress` which seemed to be a decent compromise to me

Reply:
Sounds good to me, resolving for v1